src/jobserver.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#if defined( _WIN32)
+// windows.h defines min() and max() which conflict with std::min()
+// and std::max(), which both might be used in sources (indirectly)
+// including this header. Avoid compile errors by telling windows.h
+// to not define those two.
```

Comment:
Regretably, this doesn't help is this header is included from a source file _after_ another header that previously included <windows.h> (and thus already redefined these).

The only sane way to deal with this is just to `#undef min` and `#undef max` after the include :-(

Reply:
I still think `#define NOMINMAX` is the right approach. If `windows.h` happens to be included earlier somewhere else, then we should add `#define NOMINMAX` to that location too.

The only other header currently including `windows.h` is `subprocess.h`, which isn't casuing troubles yet. We can add `#define NOMINMAX` there now, but it's not really relevant to this PR.

Reply:
How about defining `NOMINMAX` at configure-time?

```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79b99c2..66aacfa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,6 +165,12 @@ if(WIN32)
        # so that build environments which lack a C compiler, but have a C++
        # compiler may build ninja.
        set_source_files_properties(src/getopt.c PROPERTIES LANGUAGE CXX)
+
+       # windows.h defines min() and max() which conflict with std::min()
+       # and std::max(), which both might be used in sources. Avoid compile
+       # errors by telling windows.h to not define those two.
+       add_definitions(-DNOMINMAX)
 else()
        target_sources(libninja PRIVATE
                src/subprocess-posix.cc
```

In fact, `configure.py` already does this for bootstrapping...